compute/metadata/main.py

```diff
@@ -32,38 +32,46 @@
 
 def wait_for_maintenance(callback):
     url = METADATA_URL + 'instance/maintenance-event'
-    last_in_maintenance = False
+    last_maintenance_event = None
     # [START hanging_get]
-    last_etag = 0
+    last_etag = "0"
 
     while True:
         r = requests.get(
             url,
-            params={'last_etag': last_etag},
+            params={'last_etag': last_etag, 'wait_for_change': True},
             headers=METADATA_HEADERS)
 
         # During maintenance the service can return a 503, so these should
         # be retried.
         if r.status_code == 503:
             time.sleep(1)
             continue
+        elif r.status_code != requests.codes.ok:
```

Comment:
The only status the docs mention that we should explicitly check for is 503. Any other error is better surfaced through the started exception infrastructure rather than just sleeping and trying again.

Why not remove this and just use `r.raise_for_status()`?